src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IndexerAccess.cs

```diff
@@ -984,7 +984,59 @@ private BoundExpression VisitRangePatternIndexerAccess(BoundImplicitIndexerAcces
             AddPlaceholderReplacement(node.ArgumentPlaceholders[1], rangeSizeExpr);
 
             var sliceCall = (BoundCall)node.IndexerOrSliceAccess;
-            var rewrittenIndexerAccess = VisitExpression(sliceCall);
+
+            // [start..] can be simplified to .Substring(start) or .Slice(start) for some built-in types
+            // e.g. string and (ReadOnly)Span
+
+            BoundExpression? moreEfficientIndexerAccess = null;
+            if (endMakeOffsetInput is null)
+            {
+                MethodSymbol? singleArgumentOverload = null;
+                NamedTypeSymbol? typeContainingElementType = null;
+
+                if (
+                    TryGetSpecialTypeMethod(node.Syntax, SpecialMember.System_String__Substring, out var stringSubstring, isOptional: true)
+                    && sliceCall.Method.Equals(stringSubstring)
```

Comment:
Please use the `Equals` that takes a comparison kind for these calls.

Reply:
Do you think `TypeCompareKind.ConsiderEverything`  is correct?

Reply:
It is for `string`. `Span` is more interesting: we probably don't want to take tuple names into account, for example. You'll need to go through the various ignore options and add some tests around it.

Reply:
> tuple names

Do you mean the element type inside `<`...`>`?
For `Span`, I use its `UnderlyingType` to compare. I think we don't have to care about it and related ones.
`Span` is a value type, so `?` shouldn't be ignored. I found `CLRSignatureCompareOptions` and it looks great.

Reply:
As an example, what if the argument to `Slice` as a modopt on a parameter or return? That's something that would be affected by the comparison option chosen.

Reply:
Notice: This hasn't been fixed yet.

I've never heard modopt before and Google provided me little information for it.

https://stackoverflow.com/questions/69385708/what-is-a-custommodifier
https://learn.microsoft.com/ja-jp/dotnet/api/microsoft.codeanalysis.custommodifier?view=roslyn-dotnet-4.9.0

I'm afraid the above literature didn't remember me any examples.